File: src/main/java/de/hpi/tdgt/deserialisation/Deserializer.java
package de.hpi.tdgt.deserialisation;

import com.fasterxml.jackson.databind.ObjectMapper;
import de.hpi.tdgt.test.Test;
import lombok.val;

import java.io.IOException;

public class Deserializer {
    public static Test deserialize(String json) throws IOException {
        ObjectMapper mapper = new ObjectMapper();
        val test = mapper.readValue(json, Test.class);
        test.setConfigJSON(json);
        return test;
    }
}

File: src/main/java/de/hpi/tdgt/deserialisation/Deserializer.kt
package de.hpi.tdgt.deserialisation

import com.fasterxml.jackson.databind.ObjectMapper
import de.hpi.tdgt.test.Test
import java.io.IOException

object Deserializer {
    @JvmStatic
    @Throws(IOException::class)
    fun deserialize(json: String?): Test {
        val mapper = ObjectMapper()
        val test =
            mapper.readValue(json, Test::class.java)
        test.configJSON = json
        return test
    }
}